wasm-libruntime/src/workload.rs

```diff
@@ -0,0 +1,166 @@
+//! The Workload module implements retrieving and running WASM workloads.
+
+use crate::Result;
+use std::collections::HashMap;
+use wasmtime::Val;
+
+/// Struct to hold and process the workload
+pub struct Workload {
+    // TODO: think about if we assume that this buffer contains an unencrypted or encrypted workload
+    buf: Vec<u8>,
+    // TODO: more fields for tracking the state?
+}
+
+impl Workload {
+    /// Retrieves the WASM workload from an argument which implements Read
+    pub fn from_reader<R: std::io::Read>(mut r: R) -> Result<Self> {
+        let mut buf = Vec::new();
+        r.read_to_end(&mut buf)?;
+        Ok(Self { buf })
+    }
+
+    /// Get the length of the internal buffer
+    pub fn len(&self) -> usize {
+        self.buf.len()
+    }
+
+    /// Returns `true` if the workload buffer has a length 0
+    pub fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Runs a WASM workload, invoking the given function with the given arguments, returning the result vector.
+    pub fn run(&self, function_name: &str, args: &[Val]) -> Result<Vec<Val>> {
```

Comment:
Is there a standard `main()` function for WASM/WASI yet?

Reply:
According to [wasmtime](https://github.com/bytecodealliance/wasmtime/blob/177af5357880c33863cbe3363754000dbd23bddb/src/commands/run.rs#L282-L291) there's a default command export and the fallback to `_start`.